echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/config/SlackConfig.groovy

```diff
@@ -44,10 +44,24 @@ class SlackConfig {
   final static String SLACK_INCOMING_WEBHOOK = 'https://hooks.slack.com'
   final static String SLACK_CHAT_API = 'https://slack.com'
 
+  @Value('${slack.base-url:}')
+  String slackBaseUrl
+
+  @Value('${slack.force-use-incoming-webhook:false}')
+  Boolean forceUseIncomingWebhook;
+
   @Bean
   Endpoint slackEndpoint(@Qualifier("useIncomingWebHook") boolean useIncomingWebHook) {
-    log.info("Using Slack incoming webhooks: {}.", useIncomingWebHook)
-    String endpoint = useIncomingWebHook ? SLACK_INCOMING_WEBHOOK : SLACK_CHAT_API;
+    String endpoint;
+
+    if (StringUtils.isNotBlank(slackBaseUrl)) {
+      log.info("Using Slack base url: {}.", slackBaseUrl)
+      endpoint = slackBaseUrl
+    } else {
+      log.info("Using Slack incoming webhooks: {}.", useIncomingWebHook)
```

Comment:
do you think there is value to logging the actual url we will be pointing towards?.. e.g. swap the two log lines
and then 
```
log.info("Using Slack {}: {}.", useIncomingWebHook ? "incoming webhook" : "chat api", slackBaseUrl)
```

Reply:
I've implemented the logging change, we always log the url and if it's api or webhooks. Thanks for the review